ldpd/lde.c

```diff
@@ -50,6 +56,11 @@ static void		 lde_map_free(void *);
 static int		 lde_address_add(struct lde_nbr *, struct lde_addr *);
 static int		 lde_address_del(struct lde_nbr *, struct lde_addr *);
 static void		 lde_address_list_free(struct lde_nbr *);
+static void		 zclient_sync_init (u_short instance);
+static void		 lde_label_list_init(void);
+static int			 lde_get_label_chunk (void);
+static void		 on_get_label_chunk_response(uint32_t start, uint32_t end);
+static uint32_t		 lde_get_next_label(void);
 
```

Comment:
No, you can't use the same communication channel for both async and sync IO. Remember that you can get partial reads/writes when using non-blocking sockets. So, for example, if you send part of a message and then try to send a new message on top of that then all zebra will get is a corrupted stream of bytes.